### ISSUE TITLE
error pushing image: failed to push to destination

### ISSUE THREAD

**Issue by krunaladdweb** [jenkins-x/jx#6174]
### Summary 
``` 
error pushing image: failed to push to destination https://hub.docker.com/krunaladdweb/go-demo:0.0.4: Get https://https/v2/: dial tcp: lookup https on 10.100.0.10:53: no such host 
```
This is the error I am getting  at the build step in jx pipleline, 
I don't understand from where it takes the repo url, which doesn't even exist and I haven't configured it.

### Steps to reproduce the behavior
** Install jx on existing EKS cluster with `jx install`
** Create quickstart application, I used *go-http* in my case app name and repo name is "go-demo"
** Watch the pipeline activity, check for the pod created for the application to run the pipeline. in my case it is something like "krunaladdweb-go-demo-master-6s4d8-4-from-build-pack-scmnq-pod-53065f"
run the command 
`kubectl logs <podname> step-build-container-build`
and in the last line of the output, check the error.

### Expected behavior:
The docker registry url should be empty, but I have not configured.

### Actual behavior
It takes docker registry url out of nowhere

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.933
jenkins x platform 2.0.1569
Kubernetes cluster v1.14.8-eks-b7174d
kubectl            v1.16.2
helm client        Client: v2.15.1+gcf1de4f
git                2.23.0
Operating System   "Fedora release 31 (Thirty One)"
```

### Jenkins type

- [ * ] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### Kubernetes cluster

EKS Cluster created using CloudFormation

### Operating system / Environment

NAME               VERSION
jx                 2.0.933
jenkins x platform 2.0.1569
Kubernetes cluster v1.14.8-eks-b7174d
kubectl            v1.16.2
helm client        Client: v2.15.1+gcf1de4f
git                2.23.0
Operating System   "Fedora release 31 (Thirty One)"

**Comment by deanesmith** [comment]
@krunaladdweb.  @dgozalo has a `jx boot` config that will support your configuration.   We recommend using boot instead of install.

**Comment by merlineus** [comment]
@deanesmith in `jx boot` this issue also presented.

As @jstrachan mentioned in slack channel - registry should be configured as a domain name, not as an uri. But if this is not already set in `jx-requirements.yml` then `jx boot` asks external registry parameters in interactive mode and it is not possible to enter domain name there as it gives a validation error:
`Sorry, your reply was invalid: artifactory-docker.jfrog.io is not a RFC 3986 URI`
and forces user to add `https://`. So this should be fixed :)

Also `jx boot` ignores external registry credentials from interactive prompt and does not set them in `jenkins-docker-cfg`. If I add them directly to the secret all works fine till next `jx boot` run, which deletes them again. So this is another one issue - should I fill a ticket about it?

**Comment by guitcastro** [comment]
The registry should be configured using schema and not only by domain. I am trying to using the internal registry created by `jx boot` but is failing because it's not https.

**Comment by deanesmith** [comment]
Hi @merlineus,  Could you please enter an issue for that specific matter you cited?

> Also `jx boot` ignores external registry credentials from interactive prompt and does not set them in `jenkins-docker-cfg`. If I add them directly to the secret all works fine till next `jx boot` run, which deletes them again. So this is another one issue - should I fill a ticket about it?

**Comment by michaeljohnbennett** [comment]
I have the exact same issues on AKS. The prompts for a RFC url. I had to add something in, then delete it and add it on command line but it still doesn't work. 

Does anyone know the correct config for AKS otherwise this thing is a total bag of spanners... The documentation is outdated on the site, no mention of how to fix it now that `jx console` doesn't work and the UI is very immature.

I tried using `jx create docker auth ....` and that adds in ok. But I have no clue what the format for the secret update should be.
